test/tracer.spec.ts

```diff
@@ -489,6 +491,281 @@ describe('Autoblocks Tracer', () => {
+            },
+          ],
+        },
+      });
+    });
+
+    it('handles async evaluators', async () => {
+      const tracer = new AutoblocksTracer('mock-ingestion-key');
+      const { traceId } = await tracer.sendEvent('mock-message', {
+        evaluators: [new MyAsyncEvaluator()],
+      });
+
+      expect(traceId).toEqual('mock-trace-id');
+
+      expectPostRequest({
+        message: 'mock-message',
+        traceId: undefined,
+        timestamp,
+        properties: {
+          evaluations: [
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: undefined,
+              metadata: undefined,
+              evaluatorExternalId: 'my-async-evaluator',
+            },
+          ],
+        },
+      });
+    });
+
+    it('handles multiple evaluators', async () => {
+      const tracer = new AutoblocksTracer('mock-ingestion-key');
+      const { traceId } = await tracer.sendEvent('mock-message', {
+        evaluators: [
+          new MyEvaluator(),
+          new MyAsyncEvaluator(),
+          new MyFullInfoEvaluator(),
+          new MyAsyncFullInfoEvaluator(),
+        ],
+      });
+
+      expect(traceId).toEqual('mock-trace-id');
+
+      expectPostRequest({
+        message: 'mock-message',
+        traceId: undefined,
+        timestamp,
+        properties: {
+          evaluations: [
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: undefined,
+              metadata: undefined,
+              evaluatorExternalId: 'my-evaluator',
+            },
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: undefined,
+              metadata: undefined,
+              evaluatorExternalId: 'my-async-evaluator',
+            },
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: {
+                gt: 0,
+                lte: 1,
+              },
+              metadata: {
+                'some-metadata-key': 'some-metadata-value',
+              },
+              evaluatorExternalId: 'my-full-info-evaluator',
+            },
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: {
+                gt: 0,
+                lte: 1,
+              },
+              metadata: {
+                'some-metadata-key': 'some-metadata-value',
+              },
+              evaluatorExternalId: 'my-async-full-info-evaluator',
+            },
+          ],
+        },
+      });
+    });
+
+    describe('errors', () => {
+      let runEvaluatorUnsafeSpy: jest.SpyInstance;
+
+      beforeEach(() => {
+        runEvaluatorUnsafeSpy = jest
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          .spyOn(AutoblocksTracer.prototype as any, 'runEvaluatorUnsafe');
+      });
+
+      afterEach(() => {
+        runEvaluatorUnsafeSpy.mockClear();
```

Comment:
don't we want mockRestore here?

Reply:
Good catch. It doesn't really matter since I use `mockImplementationOnce`, which is why this mistake wouldn't cause side effects to other tests